web/search.dart

```diff
@@ -327,23 +420,41 @@ void initializeSearch(
   });
 
   input.addEventListener('keydown', (Event event) {
-    if (suggestionElements.isEmpty) {
-      return;
-    }
-
     if (event.type != 'keydown') {
       return;
     }
 
     event = event as KeyboardEvent;
 
+    var body = document.querySelector('body');
```

Comment:
Putting a null-assert at the end here (`document.querySelector('body')!;`) will simplify the null-checking code below.